### ISSUE TITLE
Allow A Key/Value Interface

### ISSUE THREAD

**Issue by starcraftman** [VundleVim/Vundle.vim#412]
Raised the initial issue in #403. I guess we can discuss that change here.

Gmarik suggested some key/value system for the Plugin command so that instead of json like dict we did something like;

Plugin 'somePlugin' name=newName, rtp=vim/

Thoughts? @gmarik @jdevera 

As a side note, I made a rudimentary pass previously on plugin, now on this branch:
https://github.com/starcraftman/vundle/tree/key_value

Basically, I just searched for some key/value indicator (in this example '=') then packed them in dict. Strings without the = get passed on as is.
